src/toil/batchSystems/singleMachine.py

```diff
@@ -122,19 +123,62 @@ def __init__(self, config, maxCores, maxMemory, maxDisk):
         # A pool representing the available space in bytes
         self.disk = ResourcePool(self.maxDisk, 'disk', self.acquisitionTimeout)
 
-        log.debug('Setting up the thread pool with %i workers, '
-                 'given a minimum CPU fraction of %f '
-                 'and a maximum CPU value of %i.', self.numWorkers, self.minCores, maxCores)
-        for i in range(self.numWorkers):
-            worker = Thread(target=self.worker, args=(self.inputQueue,))
-            self.workerThreads.append(worker)
-            worker.start()
+        if not self.debugWorker:
+            log.debug('Setting up the thread pool with %i workers, '
+                     'given a minimum CPU fraction of %f '
+                     'and a maximum CPU value of %i.', self.numWorkers, self.minCores, maxCores)
+            for i in range(self.numWorkers):
+                worker = Thread(target=self.worker, args=(self.inputQueue,))
+                self.workerThreads.append(worker)
+                worker.start()
+        else:
+            log.debug('Started in worker debug mode.')
+
+    def _runWorker(self, jobCommand, jobID, environment):
+        """
+        Run the jobCommand using the worker and wait for it to finish.
+        The worker is forked unless it is a '_toil_worker' job and
+        debugWorker is True.
+        """
+        startTime = time.time()  # Time job is started
+        popen = None
+        statusCode = None
+        forkWorker = not self.debugWorker or "_toil_worker" not in jobCommand
```

Comment:
We always want to fork (default behavior) unless debugworker && _toil_worker in jobcommand

```
forkWorker = not (self.debugWorker and "_toil_worker" not in jobCommand)
```

I agree that not forking when running locally could be a useful optimization, but that should be tested separately.